src/librustc_span/lib.rs

```diff
@@ -1052,12 +1065,20 @@ impl Decodable for SourceFile {
                 end_pos,
                 src: None,
                 src_hash,
-                external_src: Lock::new(ExternalSource::AbsentOk),
+                external_src: Lock::new(ExternalSource {
+                    kind: ExternalSourceKind::AbsentOk,
+                    // These values don't matter - they are never used.
+                    // The actually values are computed after this struct
+                    // is deserialized
+                    original_start_pos: BytePos(0),
```

Comment:
I'm not happy about this - it would probably be better to replace `external_src` with a `Lock<Option<ExternalSourceKind>>`. However, I don't want to make any more invasive changes until I can actually test this PR, in case I end up breaking something.